### ISSUE TITLE
[Feature] Responsive navigation in Appsmith

### ISSUE THREAD

**Issue by riodeuno** [appsmithorg/appsmith#3336]
## Summary
The navbar and the pages menu should be responsive.

## Motivation
Some apps are expected to be viewed in a mobile viewport when **_deployed_** by some users of the application.

## Additional Context

Components to be made responsive:
- Appsmith Logo
- Navbar actions
- Pages menu
- Applications list
- Share menu

Link to Figma file:
https://www.figma.com/file/LET3RLKRoqx03BMaQHkOu3/Responsive-navbar?node-id=0%3A1

**Comment by navdeepsingh** [comment]
Hello @riodeuno, Interested to solve this issue. I see there no responsiveness at all.
Like https://dev.appsmith.com/applications & Edit Page.
Can we sync, so I get better to understand the deliverables?

cc @Nikhil-Nandagopal

**Comment by riodeuno** [comment]
Hi @navdeepsingh Than you for your interest in contributing to Appsmith! We would love to have your contributions.

Unfortunately, this particular issue needs designs from the team - before we can implement this - which will need some time to be complete. The deliverables would be based on these designs. I will inform you as soon as we have the designs and this issue is ready to be picked up as an engineering task.

In the meantime, if you'd like to contribute to Appsmith, might I suggest taking a look at the issues tagged as "Good First Issues"? https://github.com/appsmithorg/appsmith/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22

**Comment by momcilo-appsmith** [comment]
@somangshu @riodeuno here is the handoff link: https://www.figma.com/file/LET3RLKRoqx03BMaQHkOu3/Responsive-navbar?node-id=0%3A1

**Comment by somangshu** [comment]
Closing this in favor of #5116
